File: BE_QuanLyTro/src/main/java/ut/edu/be_quanlytro/Dto/Request/ContractCreateManualRequest.java
package ut.edu.be_quanlytro.Dto.Request;

import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.UUID;

@Getter @Setter
public class ContractCreateManualRequest {
    private UUID roomId;
    private LocalDate startDate;
    private LocalDate endDate;
    private BigDecimal depositAmount;
    private UUID depositId;

    // Thông tin khách thuê chủ trọ tự gõ tay
    private String tenantName;
    private String tenantPhone;
    private LocalDate tenantDob;
    private String tenantHometown;
    private String tenantIdCardNumber;

    private UUID templateId;
}

File: Flutter_QuanLyTro/lib/data/models/request/contract_create_manual_request.dart
import 'dart:convert';

class ContractCreateManualRequest {
  final String roomId;
  final String startDate;
  final String endDate;
  final double depositAmount;
  final String? depositId;

  final String tenantName;
  final String tenantPhone;
  final String tenantDob;
  final String tenantHometown;
  final String tenantIdCardNumber;

  ContractCreateManualRequest({
    required this.roomId,
    required this.startDate,
    required this.endDate,
    required this.depositAmount,
    this.depositId,
    required this.tenantName,
    required this.tenantPhone,
    required this.tenantDob,
    required this.tenantHometown,
    required this.tenantIdCardNumber,
  });

  Map<String, dynamic> toMap() {
    return {
      'roomId': roomId,
      'startDate': startDate,
      'endDate': endDate,
      'depositAmount': depositAmount,
      if (depositId != null) 'depositId': depositId,
      'tenantName': tenantName,
      'tenantPhone': tenantPhone,
      'tenantDob': tenantDob,
      'tenantHometown': tenantHometown,
      'tenantIdCardNumber': tenantIdCardNumber,
    };
  }

  String toJson() => json.encode(toMap());
}